frontend/www/js/omegaup/components/arena/FeedbackCodeView.vue

```diff
@@ -0,0 +1,173 @@
+import { Vue, Component, Prop, Ref } from 'vue-property-decorator';
+import T from '../../lang';
+import CodeMirror from 'codemirror';
+import { EditorOptions, languageModeMap, modeList } from './CodeView.vue';
+import Feedback, { ArenaCourseFeedback, FeedbackStatus } from './Feedback.vue';
+
+for (const mode of modeList) {
+  require(`codemirror/mode/${mode}/${mode}.js`);
+}
+
+@Component({
+  components: {},
+})
+export default class FeedbackCodeView extends Vue {
+  @Prop() language!: string;
+  @Prop() value!: string;
+  @Prop({ default: () => new Map() }) feedbackList!: Map<
+    number,
+    ArenaCourseFeedback
+  >;
+  @Ref('cm-editor') private readonly cmEditor!: HTMLTextAreaElement;
+
+  T = T;
+  mode = languageModeMap[this.language] ?? languageModeMap['cpp17-gcc'];
+  numberOfComments = 0;
+
+  get feedbackListEmpty(): number {
+    return this.feedbackList.size;
+  }
+
+  get editorOptions(): EditorOptions {
+    return {
+      tabSize: 2,
+      lineNumbers: true,
+      mode: this.mode,
+      readOnly: true,
+      gutters: ['CodeMirror-linenumbers', 'breakpoints'],
+    };
+  }
+
+  recalculateNumberOfComments(num: number): number {
+    return num;
+  }
+
+  mounted() {
+    const editor = CodeMirror.fromTextArea(this.cmEditor, this.editorOptions);
+    let lineWidget: null | CodeMirror.LineWidget = null;
+
+    editor.on(
+      'gutterClick',
+      (codeMirror: CodeMirror.Editor, numberOfLine: number) => {
+        if (this.feedbackList.get(numberOfLine)) {
+          return;
+        }
+        lineWidget = codeMirror.addLineWidget(
+          numberOfLine,
+          showFeedbackForm(numberOfLine),
+        );
+
+        this.feedbackList.set(numberOfLine, {
+          line: numberOfLine,
+          text: null,
+          status: FeedbackStatus.New,
+        });
+        this.numberOfComments = this.recalculateNumberOfComments(
+          this.feedbackList.size,
+        );
+      },
+    );
+
+    const showFeedbackForm = (numberOfLine: number): HTMLDivElement => {
+      const marker = document.createElement('div');
+      marker.classList.add('px-2');
+
+      const feedback: ArenaCourseFeedback = {
+        text: null,
+        line: numberOfLine,
+        status: FeedbackStatus.New,
+      };
+      const componentClass = Vue.extend(Feedback);
+      const feedbackForm = new componentClass({
+        propsData: {
+          feedback,
+        },
+      });
+
+      feedbackForm.$mount();
+
+      feedbackForm.$on('submit', (feedback: ArenaCourseFeedback) => {
+        this.feedbackList.set(feedback.line, {
+          line: feedback.line,
+          text: feedback.text,
+          status: FeedbackStatus.InProgress,
+        });
+      });
+
+      feedbackForm.$on('cancel', (feedback: ArenaCourseFeedback) => {
+        this.feedbackList.delete(feedback.line);
+        if (lineWidget == null) {
+          return;
+        }
+        editor.removeLineWidget(lineWidget);
+        marker.removeChild(feedbackForm.$el);
```

Comment:
```suggestion
        marker.removeChild(feedbackForm.$el);
        feedbackForm.$destroy();
```

https://v2.vuejs.org/v2/api/#vm-destroy debería completamente destruir todo el estado interno.

Reply:
Lo había intentado, pero no me funcionó anteriormente. A ver si de esta forma, ya hace lo que debe